### DISCUSSION TITLE
Testing Avo Resource

### DISCUSSION THREAD

**Discussion by oleksandr-storchak** [avo-hq/avo#988] [Show and tell]
What is the best way to testing Avo resources?

**Comment by adrianthedev** [comment]
Hey @OlexandrStorchak,

I haven't given it a lot of thought on (external) testing. Avo is quite thoroughly tested. Mostly system tests that check the whole UI. I've seen others test it the same way, using integration testing, but I guess you could instantiate resources in unit tests and check how they behave.


I launched Avo on [Product Hunt](https://www.producthunt.com/posts/avo-for-rails) (any feedback and comments help a lot BTW) and my efforts go there today. But I'm happy to help more hands-on from tomorrow 💪

**Reply by oleksandr-storchak** [reply]
@adrianthedev 
Thank for you response! Wish success to you! 
And maybe we can see some basic example soon?

**Reply by adrianthedev** [reply]
Yes. I'll add that to the to-do list
